Trappist/src/Promact.Trappist.Test/Questions/QuestionsRepositoryTest.cs

```diff
@@ -242,6 +242,58 @@ public QuestionsRepositoryTest(Bootstrap bootstrap) : base(bootstrap)
+            ApplicationUser user = new ApplicationUser() { Email = userName, UserName = userName };
+            await _userManager.CreateAsync(user);
+            var applicationUser = await _userManager.FindByEmailAsync(user.Email);
+            //Create code-snippet Question
+            var codingQuestion = await CreateCodingQuestion();
+            await _questionRepository.AddCodeSnippetQuestionAsync(codingQuestion, applicationUser.Id);
+            //Delete code-snippet Question
+            var questionId = await _trappistDbContext.Question.Where(x => x.QuestionDetail == codingQuestion.Question.QuestionDetail).Select(x => x.Id).FirstOrDefaultAsync();
+            var questionToDelete = await _questionRepository.GetQuestionByIdAsync(questionId);
+            await _questionRepository.DeleteQuestionAsync(questionToDelete);
+            //Add single-multiple Question
+            var multipleAnswerQuestion = await CreateMultipleAnswerQuestion();
+            await _questionRepository.AddSingleMultipleAnswerQuestionAsync(multipleAnswerQuestion, applicationUser.Id);
+            //Delete single-Multiple Question
+            questionId = await _trappistDbContext.Question.Where(x => x.QuestionDetail == multipleAnswerQuestion.Question.QuestionDetail).Select(x => x.Id).FirstOrDefaultAsync();
+            questionToDelete = await _questionRepository.GetQuestionByIdAsync(questionId);
+            await _questionRepository.DeleteQuestionAsync(questionToDelete);
+            //True single-multiple & code-snippet Questions are both deleted 
+            Assert.True(_trappistDbContext.Question.Count() == 0);
+        }
+
+        /// <summary>
+        /// Method to test get Question by id
+        /// </summary>
+        [Fact]
+        public async Task GetQuestionByIdTest()
+        {
+            string userName = "user@domain.com";
+            ApplicationUser user = new ApplicationUser() { Email = userName, UserName = userName };
+            await _userManager.CreateAsync(user);
+            var applicationUser = await _userManager.FindByEmailAsync(user.Email);
+            //Create code-snippet Question
+            var codingQuestion = await CreateCodingQuestion();
+            await _questionRepository.AddCodeSnippetQuestionAsync(codingQuestion, applicationUser.Id);
+            //Get code-snippet Question
+            var questionId = await _trappistDbContext.Question.Where(x => x.QuestionDetail == codingQuestion.Question.QuestionDetail).Select(x => x.Id).FirstOrDefaultAsync();
+            //Add single-multiple Question
+            var multipleAnswerQuestion = await CreateMultipleAnswerQuestion();
+            await _questionRepository.AddSingleMultipleAnswerQuestionAsync(multipleAnswerQuestion, applicationUser.Id);
+            //Get single-Multiple Question
+            questionId = await _trappistDbContext.Question.Where(x => x.QuestionDetail == multipleAnswerQuestion.Question.QuestionDetail).Select(x => x.Id).FirstOrDefaultAsync();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable 'questionId'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1854)